### PR TITLE
Add various "make clean" targets

### PR DESCRIPTION
**Pull request by skitt**
"make clean-generated" removes generated files (by relying on the fact
that we tell git to ignore them).

"make clean-clusters" removes running clusters (this is the old "make
cleanup", which is preserved for backwards compatibility).

"make clean" cleans everything.

Signed-off-by: Stephen Kitt <skitt@redhat.com>

### CODE REVIEW

**Review by mangelajo — APPROVED** [review]

**Review by mkolesnik — APPROVED** [review]
Great idea :)

File: Makefile.inc [hunk 1]
```diff
@@ -45,6 +45,9 @@ endif
 
 # Shipyard provided targets
 
+clean:
+	git clean -X -f
+
 cleanup:
```
Comment (dfarrell07) [inline Makefile.inc]: It's a bit confusing to have both "clean" an "cleanup" targets that are totally different. Maybe we can think of some more clear names?

Comment (dfarrell07) [reply Makefile.inc]: Per discussion on the automation sync, I think the consensus was to use `make clean` for both types of cleaning and `make clean-clusters`/`make clean-git` or similar for specifically addressing the two types of cleaning. That should give a simple UX to casual users and still enable use cases for experts.

**Review by dfarrell07 — CHANGES_REQUESTED** [review]
(see inline comment)

**Review by mangelajo — APPROVED** [review]

**Comment by mangelajo** [comment]
@dfarrell07 please re-review :-)

File: Makefile.inc [hunk 2]
```diff
@@ -45,9 +45,20 @@ endif
 
 # Shipyard provided targets
 
-cleanup:
+# clean cleans everything (running clusters, generated files ...)
+clean: clean-clusters clean-git
```
Comment (mkolesnik) [inline Makefile.inc]:
```suggestion
clean: clean-clusters clean-generated
```

**Review by mkolesnik — DISMISSED** [review]
Perhaps also you can add a test for it to make sure it works as expected?

**Review by dfarrell07 — APPROVED** [review]

File: Makefile.inc [hunk 3]
```diff
@@ -45,9 +45,20 @@ endif
...
+# clean-generated removes files we generated
+clean-generated:
+	git clean -X -f
+
+# clean-clusters removes running clusters
+clean-clusters:
 	$(SCRIPTS_DIR)/cleanup.sh
 
+# cleanup is the historical clean-clusters
+cleanup: clean-clusters
```
Comment (tpantelis) [inline Makefile.inc:66]: I propose we just remove this. We could also have it just print a message that it's deprecated and then remove later but I think that would be overkill.

Comment (skitt) [reply Makefile.inc:66]: Done.

**Review by dfarrell07 — APPROVED** [review]

**Review by tpantelis — APPROVED** [review]

File: Makefile.inc (hunk 3 above)
Comment (skitt) [reply Makefile.inc:66]: It turns out other projects need this, so it’s still in for now, but I’ll remove it ASAP.

**Review by mangelajo — APPROVED** [review]
Wow, verification and all, cool :) is it time-limited to 1 minute deploy of clusters?

**Comment by skitt** [comment]
> Wow, verification and all, cool :) is it time-limited to 1 minute deploy of clusters?

I reckoned that since we do a deployment anyway, spending the time to do it again and check everything is cleaned up was OK... But we could combine the two tests!

**Comment by skitt** [comment]
@mkolesnik could you re-review?

**Pull request merged**